packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const transactionPreviewRequest: TransactionPreviewRequest = {
+        manifest: TransferTransactionManifest,
+        start_epoch_inclusive: gatewayStatusResponse.ledger_state.epoch,
+        end_epoch_exclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+        tip_percentage: 10,
+        nonce: Math.floor(Math.random() * 1000000),
+        signer_public_keys: [
+          {
+            key_type: 'EddsaEd25519',
+            key_hex: this.getRadixPrivateKey().publicKey().hex(),
+          },
+        ],
+        flags: {
+          use_free_credit: true,
+          assume_all_signature_proofs: false,
+          skip_epoch_check: true,
+        },
+      }
+      const transactionPreviewResponse: TransactionPreviewResponse =
+        await this.gatewayApiClient.transaction.innerClient.transactionPreview({
+          transactionPreviewRequest: transactionPreviewRequest,
+        })
+      const receipt = transactionPreviewResponse.receipt as {
+        fee_summary: {
+          execution_cost_units_consumed: number
+          finalization_cost_units_consumed: number
+          xrd_total_execution_cost: string
+          xrd_total_finalization_cost: string
+          xrd_total_royalty_cost: string
+          xrd_total_storage_cost: string
+          xrd_total_tipping_cost: string
+        }
+      }
+      const totalFees =
+        parseFloat(receipt.fee_summary.xrd_total_execution_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_finalization_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_royalty_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_storage_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_tipping_cost)
+      const estimatedFees: Fees = {
+        type: FeeType.FlatFee,
+        [FeeOption.Average]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fast]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fastest]: assetToBase(assetAmount(totalFees)),
+      }
+      return estimatedFees
+    } catch (error) {
+      throw new Error('Failed to calculate the fees')
+    }
+  }
+
+  getRadixNetwork(): number {
+    const network = this.getNetwork()
+    let networkId: number
+    switch (network) {
+      case Network.Mainnet:
+        networkId = NetworkId.Mainnet
+        break
+      case Network.Testnet:
+        networkId = NetworkId.InternalTestNet
+        break
+      case Network.Stagenet:
+        networkId = NetworkId.Stokenet
+        break
+      default:
+        networkId = NetworkId.Stokenet
+        break
+    }
+    return networkId
+  }
+
+  getPrivateKey(): Buffer {
+    if (!this.rootDerivationPaths) throw new Error('no root derivation paths defined')
```

Comment:
Hmm... I do not think that we should do that. We should probably use the default path for the network in this case.

Reply:
ok